charms/kserve-controller/tests/unit/test_charm.py

```diff
@@ -0,0 +1,24 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+#
+# Learn more about testing at: https://juju.is/docs/sdk/testing
+
+import unittest
+from unittest.mock import patch
+
+import yaml
+from ops.model import ActiveStatus
+from ops.testing import Harness
+
+from charm import KServeControllerCharm
+
+
+class TestCharm(unittest.TestCase):
+    def setUp(self):
+        self.harness = Harness(KServeControllerCharm)
+        self.addCleanup(self.harness.cleanup)
+
+    @patch("charm.KubernetesServicePatch", lambda x, y: None)
+    def test_pebble_ready(self):
+        self.harness.begin_with_initial_hooks()
+        self.assertEqual(self.harness.charm.unit.status, ActiveStatus())
```

Comment:
May be add test for pebble layer. I.e. grab information from pebble and confirm it is correct.
Sample:
```
pebble_plan = harness.get_container_pebble_plan("kserve-controller")
assert pebble_plan
assert pebble_plan.services
pebble_plan_info = pebble_plan.to_dict()
assert pebble_plan_info["services"]["jupyter-ui"]["command"] == "/manager --metrics-addr=:8080"
```